library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -190,6 +190,11 @@ return_status spdm_send_spdm_request(IN spdm_context_t *spdm_context,
     spdm_session_info_t *session_info;
     libspdm_session_state_t session_state;
 
+    if ((spdm_context->connection_info.capability.data_transfer_size != 0) &&
+        (request_size > spdm_context->connection_info.capability.data_transfer_size)) {
+        return RETURN_DEVICE_ERROR;
```

Comment:
I think we need return better error code, such as RETURN_BAD_BUFFER_SIZE

Reply:
For now I would say just leave it as `RETURN_DEVICE_ERROR`. It will get changed during the return status sweep.

Reply:
May I know what is problem of using RETURN_BAD_BUFFER_SIZE ?